tests/Tokenizer/Analyzer/Analysis/TypeAnalysisTest.php

```diff
@@ -123,75 +126,120 @@ public static function provideIsNullableCases(): iterable
+    {
+        yield [false, 'string|int'];
 
-            yield [true, 'null|string'];
+        yield [true, 'string|null'];
 
-            yield [true, 'string|NULL'];
+        yield [true, 'null|string'];
 
-            yield [true, 'NULL|string'];
+        yield [true, 'string|NULL'];
 
-            yield [true, 'string|int|null'];
+        yield [true, 'NULL|string'];
 
-            yield [true, 'null|string|int'];
+        yield [true, 'string|int|null'];
 
-            yield [true, 'string|null|int'];
+        yield [true, 'null|string|int'];
 
-            yield [true, 'string|int|NULL'];
+        yield [true, 'string|null|int'];
 
-            yield [true, 'NULL|string|int'];
+        yield [true, 'string|int|NULL'];
 
-            yield [true, 'string|NULL|int'];
+        yield [true, 'NULL|string|int'];
 
-            yield [false, 'string|\foo\bar'];
+        yield [true, 'string|NULL|int'];
 
-            yield [true, 'string|\foo\bar|null'];
+        yield [false, 'string|\foo\bar'];
 
-            yield [true, 'null|string|\foo\bar'];
+        yield [true, 'string|\foo\bar|null'];
 
-            yield [true, 'string|null|\foo\bar'];
+        yield [true, 'null|string|\foo\bar'];
 
-            yield [true, 'string |null| int'];
+        yield [true, 'string|null|\foo\bar'];
 
-            yield [true, 'string| null |int'];
+        yield [true, 'string |null| int'];
 
-            yield [true, 'string | null | int'];
+        yield [true, 'string| null |int'];
 
-            yield [false, 'Null2|int'];
+        yield [true, 'string | null | int'];
 
-            yield [false, 'string|Null2'];
+        yield [false, 'Null2|int'];
 
-            yield [false, 'string |Null2'];
+        yield [false, 'string|Null2'];
 
-            yield [false, 'Null2| int'];
+        yield [false, 'string |Null2'];
 
-            yield [false, 'string | Null2 | int'];
-        }
+        yield [false, 'Null2| int'];
 
-        if (\PHP_VERSION_ID >= 8_01_00) {
-            yield [false, '\foo\bar&\foo\baz'];
+        yield [false, 'string | Null2 | int'];
+    }
 
-            yield [false, '\foo\bar & \foo\baz'];
+    /**
+     * @dataProvider provideIsNullable81Cases
+     *
+     * @requires PHP 8.1
+     */
+    public function testIsNullable81(bool $expected, string $input): void
+    {
+        $analysis = new TypeAnalysis($input, 1, 2);
+        self::assertSame($expected, $analysis->isNullable());
+    }
+
+    /**
+     * @return iterable<array{bool, string}>
+     */
+    public static function provideIsNullable81Cases(): iterable
+    {
+        yield [false, '\foo\bar&\foo\baz'];
 
-            yield [false, '\foo\bar&Null2'];
-        }
+        yield [false, '\foo\bar & \foo\baz'];
 
-        if (\PHP_VERSION_ID >= 8_02_00) {
-            yield [true, '(\foo\bar&\foo\baz)|null'];
+        yield [false, '\foo\bar&Null2'];
+    }
+
+    /**
+     * @dataProvider provideIsNullable82Cases
+     *
+     * @requires PHP 8.2
```

Comment:
this could also got moved to 7581 🤷🏻

Reply:
Do you mean twice? 😉 

https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7581/files#diff-3c8f800dccefd2fcba25f19e94fe2aa6e02b4a86c55045e1cb407b46cd7b773eR220

Reply:
i assumed logic was moved, not copied 🤷🏻